lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/geo/CartesianPolygon.java

```diff
@@ -90,6 +89,40 @@ public boolean contains(CartesianPoint point) {
         return MathUtils.pnpoly(vertices.size(), verticesXValues, verticesYValues, (float) point.getX(), (float) point.getY());
     }
 
+    /**
+     * Returns true if there is a collision with another polygon.
+     *
+     * @param polygon The other polygon
+     * @return true if the polygons collide
+     */
+    public boolean isIntersectingPolygon(CartesianPolygon polygon) {
+        // Test if bounding boxes intersect
+        CartesianRectangle rectA = polygon.boundingBox;
+        CartesianRectangle rectB = calcBoundingBox(vertices);
+        if ((rectA.getA().getX() > rectB.getB().getX() || rectB.getB().getX() < rectB.getA().getX()
+                || rectA.getA().getY() < rectB.getB().getY() || rectA.getB().getY() > rectA.getA().getY())){
+            return false;
+        }
+        // Test if any polygon is completely contained within the other polygon
+        if (contains(polygon.getVertices().get(0)) || polygon.contains(vertices.get(0))) {
+            return true;
+        }
+        // Test if any edges of the polygons intersect
+        Vector3d lastVerticeP1 = vertices.get(vertices.size()-2).toVector3d();
```

Comment:
I would avoid converting to Vector3d all the time. You could just work with CartesianPoint directly. Sure, then you cannot use `Edge` anymore and you would need to move the code of `isIntersectingEdge` to this class, but you'd avoid a lot of object instantiation. You could also use classic for loops to avoid calling `vertices.get(vertices.size()-2)`:
```java
for (int vIdx1 = 1; vIdx1 < vertices.size() ; vIdx1++) {
  CartesianPoint edge1A = vertices.get(vIdx1 - 1);
  CartesianPoint edge1B = vertices.get(vIdx1);
  for (int vIdx2 = 1; vIdx2 < polygon.getVertices().size() ; vIdx2++) { 
    CartesianPoint edge2A = polygon.getVertices().get(vIdx2 - 1);
    CartesianPoint edge2B = polygon.getVertices().get(vIdx2);
    if (isEdgesIntersecting(edge1A, edge1B, edge2A, edge2B)) {
        return true;
    };

  }
}
```

for (CartesianPoint verticeP1 : vertices) {

Reply:
Suggested changes are implemented.